### PR TITLE
Fix tests

### PR DESCRIPTION
**Pull request by amCap1712**
1. Fix troi test due to change in imports.
2. Fix websockets tests as asserts from other threads don't work properly. move asserts to main thread.
3. Fix settings views test.
    Trying to rollback the transaction in the except clause can be problematic if it fails and raises an exception. As there is no exception handler for it, it will lead to the container stopping. In production, the container would restart because it is run with unless-stopped but that doesn't happen for tests. so move rollback to inside try as the first step.

### CODE REVIEW

**Comment by MonkeyDo** [comment]
I'm rooting for you @amCap1712 ! :rocket:

**Comment by amCap1712** [comment]
Per aspera ad astra.

File: listenbrainz/tests/integration/test_settings_views.py [hunk 1]
```diff
@@ -102,30 +102,30 @@ def test_delete_listens(self):
...
             self.assert200(resp)
             if json.loads(resp.data)['payload']['count'] == 0:
-                continue
-            else:
-                ok = True
+                break
 
-        if not ok:
-            self.assertEqual(json.loads(resp.data)['payload']['count'], 0)
+        print("listen count:", cache.get(REDIS_USER_LISTEN_COUNT + str(self.user['id'])))
```
Comment (mayhem) [inline listenbrainz/tests/integration/test_settings_views.py:127]: keep? delete?

**Review by mayhem — APPROVED** [review]

File: listenbrainz/tests/integration/test_settings_views.py (hunk 1 above)
Comment (amCap1712) [reply listenbrainz/tests/integration/test_settings_views.py]: removed, thanks

**Pull request merged**